pkg/raft/raft.go

```diff
@@ -670,6 +678,11 @@ func (r *raft) sendHeartbeat(to uint64) {
 	// The leader MUST NOT forward the follower's commit to
 	// an unmatched index.
 	commit := min(pr.Match, r.raftLog.committed)
+	// If all the cluster nodes are on the new version
+	// then we can safely use leader's commit index in heartbeat.
+	if r.crdbVersion != nil && r.crdbVersion.IsActive(context.Background(), clusterversion.V24_2_DisableRaftCommitIndexCapping) {
+		commit = r.raftLog.committed
+	}
```

Comment:
Can we flip this to:

```golang
commit := r.raftLog.committed

if version != nil && !version.IsActive(...) {
	commit = min(commit, pr.Match)
}
```
and rearrange the comments correspondingly?

The rationale is:
- the conditional will be simply removed in the future
- we want to test the new behaviour if the version handle is nil (in some tests)

Also, this allows removing `cluster.MakeTestingClusterSettingsWithVersions` from `raft_test.go`, as the new behaviour works in tests by default with a nil.

I think once we flip this code, we should see an (expected) change in the `lagging_commit.txt` test.